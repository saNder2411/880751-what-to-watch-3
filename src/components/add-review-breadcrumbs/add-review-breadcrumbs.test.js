import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import thunk from 'redux-thunk';

import AddReviewBreadcrumbs from './add-review-breadcrumbs.jsx';

const mockStore = configureStore([thunk]);
const store = mockStore({
  appState: {
    screen: `MAIN`,
    selectedCardId: 1,
  },
});

const title = `Titanic`;

it(`Should AddReviewBreadcrumbs render correctly`, () => {
  const markup = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <Switch>
              <Route path='/'>
                <AddReviewBreadcrumbs title={title} selectedCardId={1} />
              </Route>
            </Switch>
          </BrowserRouter>
        </Provider>
    )
    .toJSON();

  expect(markup).toMatchSnapshot();
});
