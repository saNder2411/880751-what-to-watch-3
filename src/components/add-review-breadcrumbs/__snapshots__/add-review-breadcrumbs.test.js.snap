// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddReviewBreadcrumbs render correctly 1`] = `
<nav
  className="breadcrumbs"
>
  <ul
    className="breadcrumbs__list"
  >
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
        href="/cards/1"
        onClick={[Function]}
      >
        Titanic
      </a>
    </li>
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
      >
        Add review
      </a>
    </li>
  </ul>
</nav>
`;
